### DISCUSSION TITLE
Doubled declaration of font in input placeholder

### DISCUSSION THREAD

**Discussion by loxy** [taiga-family/taiga-ui#1471] [Q&A] [answered]
Hi!

I'm wondering why the font-size is re-declared in the `textfield-placeholder()` mixin:
https://github.com/taiga-family/taiga-ui/blob/1e1d26d900cd123dddef7bcbc4bf97e64c3649f5/projects/core/styles/mixins/textfield.less#L141

Because there is a `text-basic()` applied to it, which in turn declares: `font: var(--tui-font-text-s);`. This is not a problem unless you change the value of `--tui-font-text-s` (it is declared with `0.8125rem` as default). But we use other values, derived from a base typography system, something like: `--tui-font-text-s: normal var(--text-sm) / 1.25rem var(--font-primary);`

And it also won't work if you change it directly, for example to `0.75rem`.

This is problem for all inputs, because the font size of placeholder and input are not equal in this case. Can I remove this line in a MR?

Best regards,
Kersten

**Comment by waterplea** [comment]
This is done because floating label changes font size. We change font size and not scale with transform so that it works for both traditional text and text-align: right and direction: rtl. If you want to refactor it, this still needs to be taken into consideration.

**Comment by loxy** [comment, marked as answer]
Fixed in #1494
